### ISSUE TITLE
Compilation Error

### ISSUE THREAD

**Issue by Kajanos** [rxrbln/xf86-video-ps3#1]
I've tried compiling and installing but I get a compilation error:

`In file included from ps3_cursor.c:34:
/usr/include/xorg/xf86PciInfo.h:50:2: warning: #warning "xf86PciInfo.h is deprecated.  For greater compatibility, drivers should include necessary PCI IDs locally rather than relying on this file from xorg-server." [-Wcpp]
   50 | #warning "xf86PciInfo.h is deprecated.  For greater compatibility, drivers should include necessary PCI IDs locally rather than relying on this file from xorg-server."
      |  ^~~~~~~
ps3_cursor.c: In function ‘NVSetCursorPosition’:
ps3_cursor.c:175:27: error: ‘PS3FB_IOCTL_CURSOR_POS’ undeclared (first use in this function)
  175 |     ret = ioctl(pPS3->fd, PS3FB_IOCTL_CURSOR_POS, val);
      |                           ^~~~~~~~~~~~~~~~~~~~~~
ps3_cursor.c:175:27: note: each undeclared identifier is reported only once for each function it appears in
ps3_cursor.c: In function ‘NVShowHideCursor’:
ps3_cursor.c:222:31: error: ‘PS3FB_IOCTL_CURSOR_ENABLE’ undeclared (first use in this function); did you mean ‘PS3FB_IOCTL_SCREENINFO’?
  222 |     int ret = ioctl(pPS3->fd, PS3FB_IOCTL_CURSOR_ENABLE, ShowHide);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                               PS3FB_IOCTL_SCREENINFO
ps3_cursor.c: In function ‘NVShowCursor’:
ps3_cursor.c:236:31: error: ‘PS3FB_IOCTL_CURSOR_OFFS’ undeclared (first use in this function); did you mean ‘PS3FB_IOCTL_OFF’?
  236 |     int ret = ioctl(pPS3->fd, PS3FB_IOCTL_CURSOR_OFFS, pPS3->cursor_start);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~
      |                               PS3FB_IOCTL_OFF
make[2]: *** [Makefile:512: ps3_cursor.lo] Error 1
make[2]: Opuszczenie katalogu '/root/xf86-video-ps3/src'
make[1]: *** [Makefile:448: all-recursive] Error 1
make[1]: Opuszczenie katalogu '/root/xf86-video-ps3'
make: *** [Makefile:380: all] Error 2`

**Comment by FreddyFeuerstein** [comment]
it looks like yourvdependencies are too new. maybe try downgrading them.

**Comment by FreddyFeuerstein** [comment]
i use red ribbon and ran make -j2 and it just compiled and worked.

**Comment by Kajanos** [comment]
With xorg-server 1.20.13-r1 does not work

**Comment by rxrbln** [comment]
Hi, patch welcome, or sponsorship for me to spend even more time on this ;-) https://patreon.com/renerebe

**Comment by mirh** [comment]
If I had to guess that's newer gccs having "uplifted" certain warnings to errors.

**Comment by rxrbln** [comment]
not really, the rsx driver just needs weeks of real development to make it actually fully work with newer firmware and xorg servers. 3d acceleration takes about another month or two of full time development. Which is why I started this pattern and gofundme :-)

**Comment by Kajanos** [comment]
> Hi, patch welcome, or sponsorship for me to spend even more time on this ;-) https://patreon.com/renerebe

Done ;-)

**Comment by Kajanos** [comment]
Also I have checked on FC9:
[root@PS3 xf86-video-ps3.git]# make
make  all-recursive
make[1]: Entering directory `/root/xf86-video-ps3.git'
Making all in src
make[2]: Entering directory `/root/xf86-video-ps3.git/src'
/bin/sh ../libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..    -I/usr/include/xorg -I/usr/include/pixman-1   -I/lib/modules/`uname -r`/build/include -DENABLE_NV30EXA -g -O2 -MT ps3.lo -MD -MP -MF .deps/ps3.Tpo -c -o ps3.lo ps3.c
mkdir .libs
 gcc -DHAVE_CONFIG_H -I. -I.. -I/usr/include/xorg -I/usr/include/pixman-1 -I/lib/modules/2.6.27.25-78.2.56.fc9.ppc64/build/include -DENABLE_NV30EXA -g -O2 -MT ps3.lo -MD -MP -MF .deps/ps3.Tpo -c ps3.c  -fPIC -DPIC -o .libs/ps3.o
mv -f .deps/ps3.Tpo .deps/ps3.Plo
/bin/sh ../libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..    -I/usr/include/xorg -I/usr/include/pixman-1   -I/lib/modules/`uname -r`/build/include -DENABLE_NV30EXA -g -O2 -MT ps3_gpu.lo -MD -MP -MF .deps/ps3_gpu.Tpo -c -o ps3_gpu.lo ps3_gpu.c
 gcc -DHAVE_CONFIG_H -I. -I.. -I/usr/include/xorg -I/usr/include/pixman-1 -I/lib/modules/2.6.27.25-78.2.56.fc9.ppc64/build/include -DENABLE_NV30EXA -g -O2 -MT ps3_gpu.lo -MD -MP -MF .deps/ps3_gpu.Tpo -c ps3_gpu.c  -fPIC -DPIC -o .libs/ps3_gpu.o
ps3_gpu.c: In function 'gpu_get_info':
ps3_gpu.c:82: error: storage size of 'info' isn't known
ps3_gpu.c:91: error: 'PS3FB_IOCTL_GPU_INFO' undeclared (first use in this function)
ps3_gpu.c:91: error: (Each undeclared identifier is reported only once
ps3_gpu.c:91: error: for each function it appears in.)
ps3_gpu.c: In function 'enter_direct_mode':
ps3_gpu.c:187: error: 'PS3FB_IOCTL_GPU_SETUP' undeclared (first use in this function)
make[2]: *** [ps3_gpu.lo] Error 1
make[2]: Leaving directory `/root/xf86-video-ps3.git/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/root/xf86-video-ps3.git'
make: *** [all] Error 2

**Comment by Kajanos** [comment]
On Fedora 12 PPC
[root@PS3 xf86-video-ps3]# make -j2
make  all-recursive
make[1]: Entering directory `/root/xf86-video-ps3'
Making all in src
make[2]: Entering directory `/root/xf86-video-ps3/src'
  CC     ps3.lo
  CC     ps3_cursor.lo
ps3_cursor.c: In function ‘NVSetCursorPosition’:
ps3_cursor.c:175: error: ‘PS3FB_IOCTL_CURSOR_POS’ undeclared (first use in this function)
ps3_cursor.c:175: error: (Each undeclared identifier is reported only once
ps3_cursor.c:175: error: for each function it appears in.)
ps3_cursor.c: In function ‘NVShowHideCursor’:
ps3_cursor.c:222: error: ‘PS3FB_IOCTL_CURSOR_ENABLE’ undeclared (first use in this function)
ps3_cursor.c: In function ‘NVShowCursor’:
ps3_cursor.c:236: error: ‘PS3FB_IOCTL_CURSOR_OFFS’ undeclared (first use in this function)
make[2]: *** [ps3_cursor.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
ps3.c: In function ‘PS3ScreenInit’:
ps3.c:500: warning: implicit declaration of function ‘PS3InitDma’
ps3.c:505: warning: implicit declaration of function ‘PS3ExaInit’
ps3.c:515: warning: implicit declaration of function ‘NVCursorInit’
ps3.c:535: warning: implicit declaration of function ‘PS3InitVideo’
make[2]: Leaving directory `/root/xf86-video-ps3/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/root/xf86-video-ps3'
make: *** [all] Error 2
